### PR TITLE
Lots of Oblog Additions

### PR DESCRIPTION
**Pull request by Rohan-Sreenivasan**
was able to add lots of motor inputs and encoder inputs to shuffle board through oBlog. In addition this opens an easy way to add variable to shuffleboard like the variable that counts the balls

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -78,7 +78,7 @@
      */
     @Override
     public void robotInit() {
-        Logger.configureLoggingAndConfig(this, true);
+        Logger.configureLoggingAndConfig(this, false);
```
Comment (msoucy) [inline src/main/java/frc/robot/Robot.java:82]: Don't we want separate tabs for each subsystem?

File: src/main/java/frc/robot/maps/FrancoisMap.java [hunk 2]
```diff
@@ -22,6 +22,7 @@
 import edu.wpi.first.wpilibj.AnalogTrigger;
 import edu.wpi.first.wpilibj.GyroBase;
 import edu.wpi.first.wpilibj.smartdashboard.SendableRegistry;
+import io.github.oblarg.oblog.annotations.Log;
```
Comment (msoucy) [inline src/main/java/frc/robot/maps/FrancoisMap.java]: Do you actually use this?

**Review by msoucy — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (bot190) [reply src/main/java/frc/robot/Robot.java:82]: Setting this true creates 2 tabs per subsystem, which is not ideal.

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 3]
```diff
@@ -48,6 +62,10 @@ public Drive(DifferentialDriveMap map) {
         gyro = map.getGyro();
         driveTrain = new DifferentialDrive(leftMotorGroup, rightMotorGroup);
         driveTrain.setRightSideInverted(false);
+        driveRightEncoder = rightMotorGroup.getEncoder();
+        driveLeftEncoder = leftMotorGroup.getEncoder();
+        gyroVelocity = gyro.getRate();
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Drive.java]: I realize now that this is probably already taken care of by `@Log.Gyro`.

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 4]
```diff
@@ -63,23 +81,23 @@ public CommandBase drive(DoubleSupplier forward, DoubleSupplier turn) {
             double yAxis = forward.getAsDouble();
             double xAxis = turn.getAsDouble();
             driveTrain.arcadeDrive(yAxis, xAxis);
-            SmartDashboard.putNumber("Left Drive Encoder", leftMotorGroup.getEncoder().getDistance());
-            SmartDashboard.putNumber("Right Drive Encoder", rightMotorGroup.getEncoder().getDistance());
+            SmartDashboard.putNumber("Left Drive Encoder", driveLeftEncoder.getDistance());
+            SmartDashboard.putNumber("Right Drive Encoder", driveRightEncoder.getDistance());
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Drive.java]: We shouldn't need these at all now that oblog is sending them.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 5]
```diff
@@ -25,17 +26,23 @@
...
 
-public class Indexer extends SubsystemBase {
+public class Indexer extends SubsystemBase implements Loggable {
+    @Log.SpeedController
     final SendableSpeedController singulator;
-    private final SendableSpeedController pierreMotor;
+
+    @Log.SpeedController
+    final SendableSpeedController pierreMotor;
+
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: We don't need the extra whitespace here.

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 6]
```diff
@@ -9,13 +9,16 @@
...
 // The Intake takes in the balls from the ground and it is raised and retracted by pneumatic pistons 
 // There is only one speed for the intake motor (Member: Motor) 
 // The intake interacts with the Indexer It uses motors, wheels, and pistons connected by belts and bands(The controller) 
 // The intake needs to make sure the balls pass through it and don't get jammed(Member: Motor)
 
-public class Intake extends SubsystemBase {
+public class Intake extends SubsystemBase implements Loggable {
+    @Log.SpeedController
     private final SendableSpeedController intakeMotor;
     private final IDSolenoid deployPiston;
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Intake.java:23]: We should also log the solenoid.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 7]
```diff
@@ -35,10 +37,12 @@
  * Sensors? 1) Encoders 2) Limit Switches
  */
 
-public class Lift extends SubsystemBase {
+public class Lift extends SubsystemBase implements Loggable {
 
+    @Log.SpeedController
     private PIDSpeedController elevatorMotor;
     private ISolenoid elevatorBrake;
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java:44]: Please log this solenoid as well.

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 8]
```diff
@@ -35,10 +37,12 @@
...
+    @Log.Encoder
     private IEncoder liftEncoder;
     private BooleanSupplier upperLimitSwitch;
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Lift.java:48]: We should be able to log these as well somehow. Try a plain `@Log`

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 9]
```diff
@@ -40,9 +46,10 @@ public Shooter(final RobotMap.ShooterMap map) {
         super();
         shooterHeight = map.shooterHeight();
         shooterWheelMotor = map.shooterWheel();
+        shooterEncoder = shooterWheelMotor.getEncoder();
         verticalDistance = TARGET_HEIGHT - shooterHeight;
-        SendableRegistry.add(shooterWheelMotor.getEncoder(), "Shooter");
-        SendableRegistry.enableLiveWindow(shooterWheelMotor.getEncoder());
+        SendableRegistry.add(shooterEncoder, "Shooter");
+        SendableRegistry.enableLiveWindow(shooterEncoder);
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Shooter.java]: These two lines should be removed now.

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 10]
```diff
@@ -20,8 +23,11 @@
  * or X for shoot all. Does it store any state? No. Sensors? No.
  */
 
-public class Shooter extends SubsystemBase {
+public class Shooter extends SubsystemBase implements Loggable {
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Shooter.java:25]: I'm not a big fan of it, but we probably need to add a member variable for the calculated velocity and RPM so we can log those. They should be set in the appropriate functions below.

**Review by bot190 — CHANGES_REQUESTED** [review]
This looks pretty good, a few places we need some changes though.

**Review by bot190 — APPROVED** [review]
We need to figure out how to handle the boolean suppliers, but this is a good start.

**Review by msoucy — APPROVED** [review]

**Pull request merged**